src/eduid_webapp/group_management/views/invite.py

```diff
@@ -0,0 +1,209 @@
+        graph_user = GraphUser(identifier=str(scim_user.scim_id))
+        owner_groups = current_app.scimapi_groupdb.get_groups_for_owner(owner=graph_user)
+        outgoing = get_outgoing_invites(owner_groups)
+
+    return {'incoming': incoming, 'outgoing': outgoing}
+
+
+@group_invite_views.route('/incoming', methods=['GET'])
+@MarshalWith(GroupIncomingInviteResponseSchema)
+@require_user
+def incoming_invites(user: User) -> Mapping:
+    invites = get_incoming_invites(user)
+    return {'incoming': invites}
+
+
+@group_invite_views.route('/outgoing', methods=['GET'])
+@MarshalWith(GroupOutgoingInviteResponseSchema)
+@require_user
+def outgoing_invites(user: User) -> Mapping:
+    invites = []
+    scim_user = get_scim_user_by_eppn(user.eppn)
+    if scim_user:
+        # The user can only be a group owner if there is a scim_user
+        graph_user = GraphUser(identifier=str(scim_user.scim_id))
+        owner_groups = current_app.scimapi_groupdb.get_groups_for_owner(owner=graph_user)
+        invites = get_outgoing_invites(owner_groups)
+
+    return {'outgoing': invites}
+
+
+@group_invite_views.route('/create', methods=['POST'])
+@UnmarshalWith(GroupInviteRequestSchema)
+@MarshalWith(GroupOutgoingInviteResponseSchema)
+@require_user
+def create_invite(user: User, identifier: UUID, email_address: str, role: str) -> Mapping:
+    scim_user = get_scim_user_by_eppn(user.eppn)
+    if not scim_user:
+        current_app.logger.error('User does not exist in scimapi_userdb')
+        return error_message(GroupManagementMsg.user_does_not_exist)
+
+    if not is_owner(scim_user, identifier):
+        current_app.logger.error(f'User is not owner of group with scim_id: {identifier}')
+        return error_message(GroupManagementMsg.user_not_owner)
+
+    invite_state = GroupInviteState(
+        group_scim_id=str(identifier), email_address=email_address, role=role, inviter=user.eppn
+    )
+    try:
+        current_app.invite_state_db.save(invite_state)
+    except DuplicateKeyError:
+        current_app.logger.info(
+            f'Invite for email address {invite_state.email_address} to group {invite_state.group_scim_id} '
+            f'as role {invite_state.role} already exists.'
+        )
+        return error_message(GroupManagementMsg.invite_already_exists)
+    try:
+        send_invite_email(invite_state)
+    except MailTaskFailed:
+        return error_message(CommonMsg.temp_problem)
+    current_app.stats.count(name='invite_created')
+    return outgoing_invites()
+
+
+@group_invite_views.route('/accept', methods=['POST'])
+@UnmarshalWith(GroupInviteRequestSchema)
+@MarshalWith(GroupIncomingInviteResponseSchema)
+@require_user
+def accept_invite(user: User, identifier: UUID, email_address: str, role: str) -> Mapping:
+    # Check that the current user has verified the invited email address
+    mail_addresses = [item.email for item in user.mail_addresses.to_list() if item.is_verified]
+    if email_address not in mail_addresses:
```

Comment:
I think

mail  = user.mail_addresses.find(email_address)
if mail and mail.is_verified:
   ...

is more appropriate than creating a separate representation of the data. Also, I'm unsure if we do lowercasing of email addresses somewhere, but it needs to be considered.

Reply:
`    @email.setter
    def email(self, value):
        """
        :param value: e-mail address.
        :type value: str | unicode
        """
        if not isinstance(value, string_types):
            raise UserDBValueError("Invalid 'email': {!r}".format(value))
        self._data['email'] = str(value.lower())`